tests/test_benchmarks.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+from pathlib import Path
+
+from pytest import MonkeyPatch
+
+from conda.testing.fixtures import CondaCLIFixture
+from conda.models.match_spec import MatchSpec
+from conda.common.path import get_python_short_path
+
+from conda_pypi.convert_tree import ConvertTree
+from conda_pypi.downloader import find_and_fetch, get_package_finder
+from conda_pypi.build import build_conda
+
+
+# @pytest.mark.benchmark
+# @pytest.mark.parametrize(
+#     "packages",
+#     [
+#         pytest.param(("imagesize",), id="imagesize"),  # small package, few dependencies
+#         pytest.param(("scipy",), id="scipy"),  # slightly larger package
+#         pytest.param(("jupyterlab",), id="jupyterlab"),
+#     ],
+# )
+# def test_conda_pypi_install_basic(
+#     tmp_path_factory,
+#     conda_cli: CondaCLIFixture,
+#     packages: tuple[str],
+#     benchmark,
+#     monkeypatch: MonkeyPatch,
+# ):
+#     """Benchmark basic conda pypi install functionality."""
+
+#     def setup():
+#         # Setup function is run every time. So, using benchmarks to run multiple
+#         # iterations of the test will create new paths for repo_dir and
+#         # prefix for each iteration. This ensures a clean test without any
+#         # cached packages and in a clean environment.
+#         repo_dir = tmp_path_factory.mktemp(f"{'-'.join(packages)}-pkg-repo")
+#         prefix = str(tmp_path_factory.mktemp(f"{'-'.join(packages)}"))
+
+#         monkeypatch.setattr("platformdirs.user_data_dir", lambda s: str(repo_dir))
+
+#         conda_cli("create", "--yes", "--prefix", prefix, "python=3.11")
+#         return (prefix,), {}
+
+#     def target(prefix):
+#         _, _, rc = conda_cli(
+#             "pypi",
+#             "--yes",
+#             "install",
+#             "--prefix",
+#             prefix,
+#             *packages,
+#         )
+#         return rc
+
+#     result = benchmark.pedantic(
+#         target,
+#         setup=setup,
+#         rounds=2,
+#         warmup_rounds=0,  # no warm up, cleaning the cache every time
+#     )
+#     assert result == 0
+
+
+@pytest.mark.benchmark
+@pytest.mark.parametrize(
+    "packages",
+    [
+        pytest.param(("imagesize",), id="imagesize"),  # small package, few dependencies
+        pytest.param(("jupyterlab",), id="jupyterlab"),  # large package
+        pytest.param(("numpy>2.0",), id="numpy>2.0"),  # package with version constraint
+    ],
+)
+def test_convert_tree(
+    tmp_path_factory,
+    conda_cli: CondaCLIFixture,
+    packages: tuple[str],
+    benchmark,
+):
+    """Benchmark convert_tree. This test overrides channels so the whole
+    dependency tree is converted.
+    """
+
+    def setup():
+        repo_dir = tmp_path_factory.mktemp(f"{'-'.join(packages)}-pkg-repo")
+        prefix = str(tmp_path_factory.mktemp(f"{'-'.join(packages)}"))
+        conda_cli("create", "--yes", "--prefix", prefix, "python=3.11")
```

Comment:
Usually the only reason we install Python is so conda can find site-packages. Would this test still work if we mock it instead? Something like:

https://github.com/conda-incubator/conda-pypi/blob/c9af08c9fd0418ca819cdcb6615ca4c83f752890/tests/test_plugins.py#L34-L38

Reply:
This is a good suggestion! When I tried it out I found that python is required in the target prefix in order for the package finder to work.